gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -592,6 +597,35 @@ ChannelCredentials createS2ASecuredChannelCredentials() {
     }
   }
 
+  boolean isMtlsS2AHardBoundTokensEnabled() {
+    if (!useS2A
+        // If S2A cannot be used, {@code HardBoundTokenTypes.MTLS_S2A} hard bound tokens should not
+        // be used
```

Comment:
Can we move this comment to the method level comments and mention the the reason for the 4 conditions below?

Reply:
Done in https://github.com/googleapis/sdk-platform-java/pull/3591/commits/8c060dc004f45634f9ce5c0d3a3f16a8111137c7